src/main/java/de/tum/in/www1/artemis/service/connectors/lti/Lti13Service.java

```diff
@@ -310,4 +310,20 @@ private void createOrUpdateResourceLaunch(Lti13LaunchRequest launchRequest, User
     public void buildLtiResponse(UriComponentsBuilder uriComponentsBuilder, HttpServletResponse response) {
         ltiService.buildLtiResponse(uriComponentsBuilder, response);
     }
+
+    public void buildLtiEmailInUseResponse(HttpServletResponse response, OidcIdToken ltiIdToken) {
+        Optional<String> optionalUsername = artemisAuthenticationProvider.getUsernameForEmail(ltiIdToken.getEmail());
+
+        String sanitizedUsername = getSanitizedUsername(optionalUsername.get());
```

Comment:
shouldn't we check if optionalUsername is present before getting the value?

Reply:
Thank you for the heads up, you are right. I fixed that now.